--- a/Cron/Webhook.php
+++ b/Cron/Webhook.php
@@ -198,8 +198,14 @@ class Webhook
                 /**
                  * @todo change the merge vars
                  */
-                $customer->setFirstname($fname);
-                $customer->setLastname($lname);
+                $fname = trim($fname);
+                $lname = trim($lname);
+                if ($fname != "") {
+                    $customer->setFirstname($fname);
+                }
+                if ($lname != "") {
+                    $customer->setLastname($lname);
+                }
                 $customer->getResource()->save($customer);
             }
         } else {
--